include/seqan3/range/view/take.hpp

```diff
@@ -362,23 +372,32 @@ class view_take : public ranges::view_interface<view_take<urng_t, exactly, or_th
      *
      * No-throw guarantee.
      */
-    constexpr iterator begin() noexcept
+    constexpr auto begin() noexcept
     {
-        return {seqan3::begin(urange), 0, target_size, &(*this)};
+        if constexpr (std::ranges::RandomAccessRange<urng_t> && std::ranges::SizedRange<urng_t>)
+            return std::ranges::begin(urange);
+        else
+            return iterator{seqan3::begin(urange), 0, target_size, &(*this)};
```

Comment:
is `&(*this)` not the same as `this`? 🤔

Reply:
don't use seqan3::begin

Reply:
> is &(*this) not the same as this? thinking

:thinking:  I copy'n'pasted it from before.